### PR TITLE
hardfault_log:Add Fault Status registers

### PR DESCRIPTION
**Pull request by davids5**
Add the ARM Fault Status registers

```
	cfsr;      0x0d28 Configurable fault status register 
	hfsr;      0x0d2c Hard fault status register 
	dfsr;      0x0d30 Debug fault status register 
	mmfsr;  0x0d34 Mem manage address register 
	bfsr;      0x0d38 Bus fault address register 
	afsr;      0x0d3c Auxiliary fault status register 
and on m7
	abfsr; 0x0fa8 Auxiliary Bus Fault Status Register 
```

### CODE REVIEW

File: src/systemcmds/hardfault_log/hardfault_log.c [hunk 1]
```diff
@@ -392,6 +392,31 @@ static int write_registers(uint32_t regs[], char *buffer, int max, int fd)
 	return OK;
 }
 
+/****************************************************************************
+ * write_registers
+ ****************************************************************************/
+static int write_fault_registers(fault_regs_s fault_regs, char *buffer, int max, int fd)
```
Comment (bkueng) [inline src/systemcmds/hardfault_log/hardfault_log.c:398]:
```suggestion
static int write_fault_registers(fault_regs_s *fault_regs, char *buffer, int max, int fd)
```

Comment (davids5) [reply src/systemcmds/hardfault_log/hardfault_log.c]: @bkueng - Done Thanks!

**Comment by davids5** [comment]
@bkueng FYI I had the wrong had CONFIG_ARCH_ARMV7M -> CONFIG_ARCH_CORTEXM7

7M M7 looked the same to me :)

**Review by bkueng — APPROVED** [review]

**Comment by bkueng** [comment]
The NXP builds fail (nxp_fmuk66-v3_default):
```
 ../../platforms/nuttx/src/px4/common/board_crashdump.c: In function 'board_crashdump':
../../platforms/nuttx/src/px4/common/board_crashdump.c:291:33: error: implicit declaration of function 'getreg32'; did you mean 'setregid'? [-Werror=implicit-function-declaration]
  pdump->info.fault_regs.cfsr  = getreg32(NVIC_CFAULTS);
                                 ^~~~~~~~
                                 setregid
```

**Review by bkueng — APPROVED** [review]

**Pull request merged**